src/NHibernate/Async/Transaction/AdoNetWithDistributedTransactionFactory.cs

```diff
@@ -0,0 +1,40 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by AsyncGenerator.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+
+using System;
+using System.Collections;
+using System.Transactions;
+using NHibernate.Engine;
+using NHibernate.Engine.Transaction;
+using NHibernate.Impl;
+
+namespace NHibernate.Transaction
+{
+	using System.Threading.Tasks;
+	using System.Threading;
+	/// <content>
+	/// Contains generated async methods
+	/// </content>
+	public partial class AdoNetWithDistributedTransactionFactory : ITransactionFactory
+	{
+
+		public async Task ExecuteWorkInIsolationAsync(ISessionImplementor session, IIsolatedWork work, bool transacted, CancellationToken cancellationToken)
+		{
+			cancellationToken.ThrowIfCancellationRequested();
+			using (var tx = new TransactionScope(TransactionScopeOption.Suppress,TransactionScopeAsyncFlowOption.Enabled))
```

Comment:
[Fixed](https://github.com/maca88/AsyncGenerator/commit/c5e65a5f1052a9f5d690d9a7a50a00c0d70e7c31). Also, the plugin for adding the AsyncFlowOption was moved into the library itself.